File: kotlin/src/main/kotlin/application/aoc2023/DayTwo.kt
package application.aoc2023

import model.BallGame
import utils.TextFileProcessor

class DayTwo {
    private val fileLines = TextFileProcessor(2023, 2).getFileLines()

    fun solvePartA() {
        var idTotal = 0
        for (gameLine in fileLines) {
            val game = BallGame(gameLine, 12, 13, 14)

            if (game.isPossible()) {
                idTotal += game.getGameId()
            }
        }
        println("Part A: $idTotal")
    }

    fun solvePartB() {
        var powerTotal = 0
        for (gameLine in fileLines) {
            val game = BallGame(gameLine, 12, 13, 14)

            powerTotal += game.getGamePowerMinimum()
        }
        println("Part B: $powerTotal")
    }
}


fun main() {
    DayThree().solvePartA()
    DayThree().solvePartB()
}

File: python/src/main/python/application/aoc2023/day_two.py
from src.main.python.model.ball_game import BallGame
from src.main.python.utils.text_file_processor import TextFileProcessor


class DayTwo:
    __file_lines: list[str] = TextFileProcessor(2023, 2).get_file_lines()

    def solve_part_a(self) -> None:
        id_total: int = 0

        for game_line in self.__file_lines:
            game_line = game_line.replace("\n", "")

            game: BallGame = BallGame(game_line, 12, 13, 14)

            if game.is_possible():
                id_total += game.get_game_id()

        print(f"Part A: {id_total}")

    def solve_part_b(self) -> None:
        power_total: int = 0

        for game_line in self.__file_lines:
            game_line = game_line.replace("\n", "")

            game: BallGame = BallGame(game_line, 12, 13, 14)

            power_total += game.get_game_power_minimum()

        print(f"Part B: {power_total}")


if __name__ == "__main__":
    DayTwo().solve_part_a()
    DayTwo().solve_part_b()
